src/openlcb/BroadcastTimeDefs.hxx

```diff
@@ -0,0 +1,346 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @file BroadcastTimeDefs.hxx
+ *
+ * Static definitions for implementations of the OpenLCB Broadcast Time
+ * Protocol.
+ *
+ * @author Stuart W. Baker
+ * @date 14 October 2018
+ */
+
+#ifndef _OPENLCB_BROADCASTTIMEDEFS_HXX_
+#define _OPENLCB_BROADCASTTIMEDEFS_HXX_
+
+#include <time.h>
+
+#include "openlcb/Defs.hxx"
+
+namespace openlcb
+{
+
+/// Static constants and helper functions for Broadcast Time Protocol.
+struct BroadcastTimeDefs
+{
+    /// Unique identifier for the Default Fast Clock
+    static constexpr NodeID DEFAULT_FAST_CLOCK_ID = 0x010100000100ULL;
+
+    /// Unique identifier for the Default Real-Time Clock
+    static constexpr NodeID DEFAULT_REALTIME_CLOCK_ID = 0x010100000101ULL;
+
+    /// Unique identifier for Alternate Clock 1
+    static constexpr NodeID ALTERNATE_CLOCK_1_ID = 0x010100000102ULL;
+
+    /// Unique identifier for Alternate Clock 2
+    static constexpr NodeID ALTERNATE_CLOCK_2_ID = 0x010100000103ULL;
+
+    /// Type of event
+    enum EventType
+    {
+        REPORT_TIME = 0, ///< report time event
+        REPORT_DATE,     ///< report date event
+        REPORT_YEAR,     ///< report year event
+        REPORT_RATE,     ///< report rate event
+        SET_TIME,        ///< set time event
+        SET_DATE,        ///< set date event
+        SET_YEAR,        ///< set year event
+        SET_RATE,        ///< set rate event
+        QUERY,           ///< query event
+        STOP,            ///< stop clock event
+        START,           ///< start clock event
+        DATE_ROLLOVER,  ///< date rollover event
+        UNDEFINED,       ///< undefined event
+    };
+
+    enum
+    {
+        EVENT_ID_MASK      = 0xFFFFFFFFFFFF0000, ///< Unique ID mask
+        EVENT_SUFFIX_MASK  = 0x000000000000FFFF, ///< suffix mask
+        EVENT_TYPE_MASK    = 0x000000000000F000, ///< type mask
+        EVENT_HOURS_MASK   = 0x0000000000001F00, ///< hours mask
+        EVENT_MINUTES_MASK = 0x00000000000000FF, ///< minutes mask
+        EVENT_MONTH_MASK   = 0x0000000000000F00, ///< month mask
+        EVENT_DAY_MASK     = 0x00000000000000FF, ///< day mask
+        EVENT_YEAR_MASK    = 0x0000000000000FFF, ///< rate mask
+        EVENT_RATE_MASK    = 0x0000000000000FFF, ///< rate mask
```

Comment:
I think it might make sense to stop the anonymous enum after this and start again for the remaining entries, because above here we have 64-bit entries and below only 16 to 32 bit entries. I would guess now the symbols below also have 64-bit storage class due to being in the same enum.